L1Trigger/TrackTrigger/src/TTStubAlgorithm_official.cc

```diff
@@ -64,65 +64,80 @@ void TTStubAlgorithm_official< Ref_Phase2TrackerDigi_ >::PatternHitCorrelation(
+  ///    be consistent with the definition given above 
+  
+  double offsetD = 2 * delta * ( mp0.x() - (top0->nrows()/2 - 0.5) ) * (pitch0.first / pitch1.first); /// In HALF-STRIP units!
+  int offsetI = ((offsetD>0)-(offsetD<0))*floor(fabs(offsetD)); /// In HALF-STRIP units!
+
   if (stDetId.subdetId()==StripSubdetector::TOB)
   {
-    int window = 2*barrelCut.at( theTrackerTopo_->layer(stDetId) );
-    /// POSITION IN TERMS OF PITCH MULTIPLES:
-    ///       0 1 2 3 4 5 5 6 8 9 ...
-    /// COORD: 0 1 2 3 4 5 6 7 8 9 ...
-    /// OUT   | | | | | |x| | | | | | | | | |
-    ///
-    /// IN    | | | |x|x| | | | | | | | | | |
-    ///             THIS is 3.5 (COORD) and 4.0 (POS)
-    /// 1) disp is the difference between average row coordinates
-    ///    in inner and outer stack member, in terms of outer member pitch
-    ///    (in case they are the same, this is just a plain coordinate difference)
-    double dispD = 2 * (mp1.x() - mp0.x()) * (pitch0.first / pitch1.first); /// In HALF-STRIP units!
-    int dispI = ((dispD>0)-(dispD<0))*floor(fabs(dispD)); /// In HALF-STRIP units!
-    /// 2) offset is the projection with a straight line of the innermost
-    ///    hit towards the ourermost stack member, still in terms of outer member pitch
-    ///    NOTE: in terms of coordinates, the center of the module is at NROWS/2-0.5 to
-    ///    be consistent with the definition given above 
-    double offsetD = 2 * DR/R0 * ( mp0.x() - (top0->nrows()/2 - 0.5) ) * (pitch0.first / pitch1.first); /// In HALF-STRIP units!
-    int offsetI = ((offsetD>0)-(offsetD<0))*floor(fabs(offsetD)); /// In HALF-STRIP units!
-
-    /// Accept the stub if the post-offset correction displacement is smaller than the half-window
-    if ( fabs(dispI - offsetI) <= window ) /// In HALF-STRIP units!
+    int layer  = theTrackerTopo_->layer(stDetId);
+    int ladder = theTrackerTopo_->tobRod(stDetId);
+
+    window = 2*barrelCut.at( layer );
+
+    // This calculation is OK for the non-tilted part, if we are in the tilted detector this is different
+
+    if (m_tilted)
     {
-        aConfirmation = true;
-        aDisplacement = dispI; /// In HALF-STRIP units!
-        anOffset = offsetI; /// In HALF-STRIP units!
-    } /// End of stub is accepted
+      if (layer==1 && fabs(det0->position().z())>15)
+      {
+	ladder = 6-(6-ladder)*fabs(det0->position().z())/det0->position().z();
+	window = 2*(tiltedCut.at(layer)).at(ladder);
+      }
+
+      if (layer==2 && fabs(det0->position().z())>25)
+      {
+	ladder = 6.5-(6.5-ladder)*fabs(det0->position().z())/det0->position().z();
+	window = 2*(tiltedCut.at(layer)).at(ladder);
+      }
+
+      if (layer==3 && fabs(det0->position().z())>34)
+      {
+	ladder = 7-(7-ladder)*fabs(det0->position().z())/det0->position().z();
+	window = 2*(tiltedCut.at(layer)).at(ladder);
+      }
+    }
+
+ 
   }
   else if (stDetId.subdetId()==StripSubdetector::TID)
```

Comment:
I know that this one is not clean. Now, is there a method from trackertopology telling if a barrel module is tilted or not. I know how to get that from the detid, but I don't know if a method is doing that. 

Then, for the ladder number I need to how many tilted rings are on the layer. Now, this is certainly something we can pass as an option. 

So there is certainly some way to remove hard code, but I need a method extracting the barrel module type from the detid.